File: androidx/SDKSampleJava/app/src/main/java/com/gyant/sdksamplejava/DisplayGyantViewWithThemeActivity.java
package com.gyant.sdksamplejava;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import android.widget.FrameLayout;

import com.gyant.chat_sdk.GyantChat;
import com.gyant.chat_sdk.GyantView;

import java.util.HashMap;
import java.util.Map;

public class DisplayGyantViewWithThemeActivity extends AppCompatActivity {

    GyantView gyantView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_display_gyant_view);

        Map<String, String> botPalette = new HashMap<String, String>();
        botPalette.put("primaryColor1","ff0000");

        Map<String, String> providerPalette = new HashMap<String, String>();
        botPalette.put("primaryColor1","00ff00");


        Map<String, Map<String, String>> themeMap = new HashMap<String, Map<String, String>>();
        themeMap.put("bot", botPalette);
        themeMap.put("provider", providerPalette);

        GyantChat gyantChat = GyantChat.getInstance()
                .clientId("client_id")
                .patientId("patient_id")
                .isDev(true)
                .withTheme(themeMap).start();

        this.gyantView = gyantChat.createView(this, getLifecycle());

        FrameLayout frameLayout = (FrameLayout) findViewById(R.id.frame_layout);
        frameLayout.addView(gyantView);
    }


    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        this.gyantView.onRequestPermissionsResult(requestCode, permissions, grantResults);
    }
}

File: androidx/SDKSampleKotlin/app/src/main/java/com/gyant/sdksamplekotlin/DisplayGyantViewWithThemeActivity.kt
package com.gyant.sdksamplekotlin

import android.os.Bundle
import androidx.annotation.NonNull
import androidx.appcompat.app.AppCompatActivity
import android.widget.FrameLayout

import com.gyant.chat_sdk.GyantChat
import com.gyant.chat_sdk.GyantView

import java.util.HashMap


class DisplayGyantViewWithThemeActivity : AppCompatActivity() {
    private lateinit var gyantView: GyantView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_display_gyant_view)

        val botPalette = HashMap<String, String>()
        botPalette["primaryColor1"] = "ff0000"

        val providerPalette = HashMap<String, String>()
        botPalette["primaryColor1"] = "00ff00"


        val themeMap = HashMap<String, Map<String, String>>()
        themeMap["bot"] = botPalette
        themeMap["provider"] = providerPalette

        val gyantChat = GyantChat.getInstance()
            .clientId("client_id")
            .patientId("patient_id")
            .withTheme(themeMap)
            .isDev(true)
            .start()

        this.gyantView = gyantChat.createView(this, lifecycle)

        val frameLayout = findViewById<FrameLayout>(R.id.frame_layout)
        frameLayout.addView(gyantView)
    }

    override fun onRequestPermissionsResult(requestCode: Int, @NonNull permissions: Array<String>, @NonNull grantResults: IntArray) {
        this.gyantView.onRequestPermissionsResult(requestCode, permissions, grantResults)
    }
}
